### PR TITLE
feat(picker): add ability to use picker inline

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: N/A
Picker can only be presented through `pickerController`.

## What is the new behavior?

- Picker can now be used as an inline component, with support for both `isOpen` and `trigger`.
- Also fixes an issue unique to inline pickers where the selected item did not receive the selected class when the picker is re-opened.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/picker/picker.tsx [hunk 1]
```diff
@@ -114,16 +165,63 @@ export class Picker implements ComponentInterface, OverlayInterface {
    */
   @Event({ eventName: 'ionPickerDidDismiss' }) didDismiss!: EventEmitter<OverlayEventDetail>;
 
+  /**
+   * Emitted after the picker has presented.
+   * Shorthand for ionAlertWillDismiss.
```
Comment (sean-perkins) [inline core/src/components/picker/picker.tsx:170]:
```suggestion
   * Shorthand for ionPickerWillDismiss.
```

File: core/src/components/picker/picker.tsx [hunk 2]
```diff
@@ -114,16 +165,63 @@ export class Picker implements ComponentInterface, OverlayInterface {
...
+   */
+  @Event({ eventName: 'didPresent' }) didPresentShorthand!: EventEmitter<void>;
+
+  /**
+   * Emitted before the picker has presented.
+   * Shorthand for ionAlertWillPresent.
```
Comment (sean-perkins) [inline core/src/components/picker/picker.tsx:176]:
```suggestion
   * Shorthand for ionPickerWillPresent.
```

File: core/src/components/picker/picker.tsx [hunk 3]
```diff
@@ -114,16 +165,63 @@ export class Picker implements ComponentInterface, OverlayInterface {
...
+   */
+  @Event({ eventName: 'willPresent' }) willPresentShorthand!: EventEmitter<void>;
+
+  /**
+   * Emitted before the picker has dismissed.
+   * Shorthand for ionAlertWillDismiss.
```
Comment (sean-perkins) [inline core/src/components/picker/picker.tsx:182]:
```suggestion
   * Shorthand for ionPickerWillDismiss.
```

**Review by sean-perkins — CHANGES_REQUESTED** [review]
Looks good besides a few copy & paste comments. Will need to rebuild once those changes are in.

**Comment by liamdebeasi** [comment]
Need to merge https://github.com/ionic-team/ionic-framework/pull/26333 first so I can add tests for inline picker

**Review by sean-perkins — APPROVED** [review]
Nice work!

**Pull request merged**